src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/eltwise_caching.cpp

```diff
@@ -499,6 +499,7 @@ INSTANTIATE_TEST_SUITE_P(smoke_EltwiseCache_3D_planar_dyn, EltwiseCacheTest,
                                 ::testing::Values(cpuParams_3D_planar)),
                         EltwiseCacheTest::getTestCaseName);
 
+#if !defined(OPENVINO_ARCH_RISCV64)
```

Comment:
To insert `#if !defined(OPENVINO_ARCH_RISCV64)` everywhere - is really bad idea and not scalable solution.
Firstly, blocked and per-channel layouts are supported by default riscv64 Eltwise implementation - reference. Only SHL doesn't support these layouts.
Secondly, our test infrastructure provides filter for CPU-specific test parameters - [filterCPUInfo](https://github.com/openvinotoolkit/openvino/blob/6805e6229add575fdae0f4a5d90e70713924bbfe/src/plugins/intel_cpu/tests/functional/utils/filter_cpu_info.hpp#L11):
https://github.com/openvinotoolkit/openvino/blob/2e8023faa0c8b92150255920ede4067728f547e6/src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/eltwise_caching.cpp#L564

[x64](https://github.com/openvinotoolkit/openvino/blob/2e8023faa0c8b92150255920ede4067728f547e6/src/plugins/intel_cpu/tests/functional/utils/x64/filter_cpu_info.cpp) and [aarch64](https://github.com/openvinotoolkit/openvino/blob/2e8023faa0c8b92150255920ede4067728f547e6/src/plugins/intel_cpu/tests/functional/utils/arm/filter_cpu_info.cpp) architectures have own implementations of this filter. The idea is to implement riscv64-specific filter as well. `filterCPUInfo` includes the following functions:
1. `filterCPUInfoForArch` - we should leave only `ref` and `shl` primitives
2. `filterCPUInfoForDevice` - should be as for aarch64 - skip all x64-specific ISA
3. `filterCPUInfoForDeviceWithFP16` - return empty collection as aarch64
4. `filterCPUSpecificParams` - we can leave only planar layouts for riscv64. At the moment, all operations in models are executed in planar layouts (`nch, nchw, ncdhw`).

After that you don't need almost anymore insert `#ifdef` to disable per-channel and blocked layout testing

Also note please that you need to add new riscv64-specific utils directory to `exclude_files` in [CMakeLists.txt](https://github.com/openvinotoolkit/openvino/blob/6a945598ec14ebc299333140e084452e447c37e3/src/plugins/intel_cpu/tests/functional/CMakeLists.txt#L42) file for non riscv64 platforms.

Reply:
Thanks for these detailed explanation! I will try to implement this and remove all the redundant macros!